EX2Linux.cpp

```diff
@@ -56,58 +67,142 @@ void freeArgs(std::vector<char*>& args) {
 
 //------------------------------------------------------------------------------------------
 /**
- * this function recieves the first cell of the vector char* which is the command, a vector of 
- * arguments which represent the arguments, a boolean value to see if the command must run on 
- * bg or fg, and a vector of pid_t whihc will store all processes ran in bg. the function's 
- * goal is to execute the command, the execvp uses PATHS to search for the command and executes it
- * if the path is not found an error is displayed to user, it first creates a process:
- * CHILD PROCESS: uses the execvp function to execute the command with the following arguments 
- * PARENT PROCESS: checks first if the process must run in bg or fg, if fg, it waits for the 
- *                  child process to finish, if bg, it does not need to wait the process keeps
- *                  running on bg, and appends the process to the vector of bgprocesses.
- * the function as well handles errors if any failures occur.
-*/
+ * This function receives the command (first element of the arguments vector), a vector of
+ * arguments, a boolean value to determine if the command must run in the background or foreground,
+ * and a vector of pid_t to store background processes.
+ * The function executes the command using the execvp function. It handles input and output
+ * redirection if requested by the user.
+ * -------- PART 2 --------
+ * Re-implemented this function so it can allow the user to use redirections and pipes as well.
+ * The child process now checks if the args vector consists of any of these operators '<' '>'
+ * '|' and accordingly uses the required system calls to fullfil the requirements if the found operator.
+ */
+
 void executeCommand(const std::string& command, std::vector<char*>& args, bool isBackground,
-                    std::vector<pid_t>& backgroundProcesses) {
-    pid_t pid = fork();
-    if(pid == 0){
-        //CHILD
-        if (execvp(command.c_str(), args.data()) == -1) {
-            perror("Command not found.");
-            exit(1);
+    std::vector<pid_t>& backgroundProcesses) {
+    try {
+        pid_t pid = fork();
+        if (pid == 0) {
+            // CHILD
+            int inputFd = -1;
+            auto inputIt = std::find_if(args.begin(), args.end(),
```

Comment:
This logic is duplicated. This is a sign that you should separate it to a function